core/connection/manager.go

```diff
@@ -166,6 +185,8 @@ func (manager *connectionManager) startConnection(consumerID identity.Identity,
 		Proposal:      proposal,
 	}
 
+	manager.statsSender = manager.statsSenderFactory(consumerID, sessionID, proposal, time.Minute)
```

Comment:
This stats sender lifecycle is still strage to me. It's constructed on every connection, then on state callback its start/stop methods are called. So its temporal depedency - created on one method, methods called on different. Can that stats sender be made like component injected into connection manager and simply handle stats, and conn start stop .methods internally (by resetting itself when needed or doing separate unrelated goroutines whatever)

Reply:
I agree that it's strange. I think this component shouldn't even be a component, a func would do here. There's no state it needs to keep internally - just send a request with the given data. Maybe it needs to be merged with the keeper, just adding a .Send() to it and renaming it to a stat handler? I don't feel that the schedule part of the component makes it a worthy component at all. I'm also not entirely sure that this is the appropriate PR to rewrite this whole component in general - the idea here was to move it up a level, instead of having it under the openvpn connection.